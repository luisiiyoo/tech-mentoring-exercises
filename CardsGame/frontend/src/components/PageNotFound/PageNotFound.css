.PageNotFound {
  height: inherit;
  width: inherit;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  align-items: center;
}
