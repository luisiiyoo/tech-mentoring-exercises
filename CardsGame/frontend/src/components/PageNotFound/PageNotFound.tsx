import React from 'react';
import './PageNotFound.css';
import ErrorDisplay from '../ErrorDisplay';

const PageNotFound: React.FC = () => (
  <div className="PageNotFound" data-testid="PageNotFound">
    <ErrorDisplay message={`Page not found`} statusCode={404} />
  </div>
);

export default PageNotFound;
