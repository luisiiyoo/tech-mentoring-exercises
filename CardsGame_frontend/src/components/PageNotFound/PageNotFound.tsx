import React from 'react';
import './PageNotFound.css';

const PageNotFound: React.FC = () => (
  <div className="PageNotFound">
    <h2>404 Not Found</h2>
  </div>
);

export default PageNotFound;
