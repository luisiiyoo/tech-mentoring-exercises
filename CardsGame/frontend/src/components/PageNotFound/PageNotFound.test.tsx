import React from 'react';
import { cleanup, render } from '@testing-library/react';
import PageNotFound from './PageNotFound';

describe('PageNotFound', () => {
  afterEach(cleanup);

  it('should take a snapshot', () => {
    const { asFragment } = render(<PageNotFound />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render a not found message', () => {
    const { getByTestId } = render(<PageNotFound />);
    expect(getByTestId('PageNotFound')).toHaveTextContent(
      'Error 404 Page not found',
    );
  });
});
