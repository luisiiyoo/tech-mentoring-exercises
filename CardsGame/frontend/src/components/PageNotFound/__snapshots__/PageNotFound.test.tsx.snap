// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="PageNotFound"
    data-testid="PageNotFound"
  >
    <div
      class="Error"
    >
      <div
        class="Error-Label"
        data-testid="ErrorLabel"
      >
        <span>
           Error 404 
        </span>
      </div>
      <div
        class="Error-Message"
        data-testid="ErrorMessage"
      >
        Page not found
      </div>
    </div>
  </div>
</DocumentFragment>
`;
